updater/internal/rollback/rollback.go

```diff
@@ -29,15 +29,20 @@ import (
 	"go.uber.org/zap"
 )
 
-// ActionAppender is an interface that allows actions to be appended to it.
-//go:generate mockery --name ActionAppender --filename action_appender.go
-type ActionAppender interface {
+//Rollbacker is an interface that performs rollback/backup actions.
+//go:generate mockery --name Rollbacker --filename rollbacker.go
+type Rollbacker interface {
+	// AppendAction saves the action so that it can be rolled back later.
 	AppendAction(action action.RollbackableAction)
+	// Backup backs up the current installation
+	Backup() error
+	// Rollback undoes the actions recorded by AppendAction.
+	Rollback()
 }
 
-// Rollbacker is a struct that records rollback information,
+// rollbacker is a struct that records rollback information,
 // and can use that information to perform a rollback.
-type Rollbacker struct {
+type rollbacker struct {
```

Comment:
Same here having this just be a lowercase name of the interface could lead to confusion.